src/SourceBuild/content/eng/tools/BinaryToolKit/BinaryTool.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Logging;
+
+namespace BinaryToolKit;
+
+public partial class BinaryTool
+{
+
+    // Target directory - where to look for binaries
+    private readonly string TargetDirectory;
+
+    // Allowed binaries file - known binaries in the target directory that should not be removed. Needed for validation and cleaning.
+    private readonly string? AllowedBinariesFile;
+
+    // Disallowed source build binaries file - known binaries in the target directory that are disallowed for source-build. Needed for validation.
+    private readonly string? DisallowedSbBinariesFile;
+
+    // Updated allowed binaries file - updated baseline of allowed binaries
+    private readonly string? UpdatedAllowedBinariesFile;
+
+    // Updated disallowed source build binaries file - updated baseline of disallowed source build binaries
+    private readonly string? UpdatedDisallowedSbBinariesFile;
+
+    // New binaries file - list of new binaries detected in the target directory
+    private readonly string? NewBinariesFile;
+
+    // Mode to run the binary detection in
+    private Mode.ModeOptions ModeOption;
+
+    // Logger
+    private ILogger Log;
+
+    public BinaryTool(string targetDirectory, string outputReportDirectory, string? allowedBinariesFile, string? disallowedSbBinariesFile, Mode.ModeOptions mode)
+    {
+        ModeOption = Mode.GetFullMode(mode);
+        Log = ConfigureLogger();
+
+        TargetDirectory = GetAndValidateFullPath(targetDirectory, true, false)!;
```

Comment:
Nit: For callers of this method it might be more readable to use named parameters: `(targetDirectory, isDirectory: true, createIfNotExist: false)`